CDCgov/prime-reportstream
# 16394 add ack functionality

This PR adds ACK functionality to the waters and report submission endpoints.

Test Steps:
1. Run ReportStream
2. Submit the following HL7. Ensure the content type header has been specified as "application/hl7-v2"
```
MSH|^~\&|Epic|Hospital|LIMS|StatePHL|20241003000000||ORM^O01^ORM_O01|4AFA57FE-D41D-4631-9500-286AAAF797E4|T|2.5.1|||AL|NE 
```
3. The response body should be HL7

## Changes
- New ACK functionality
- Ability to pick apart more MSH header segments

## Checklist

### Testing

## Linked Issues
- Fixes #16394
